frontend/src/features/collections/collection-replay-dialog.ts

```diff
@@ -0,0 +1,387 @@
+                .homeUrl=${this.homeUrl}
+                .homeTs=${this.homeTs}
+                @btrix-select=${async (
+                  e: CustomEvent<SelectSnapshotDetail>,
+                ) => {
+                  this.selectedSnapshot = e.detail.item;
+                }}
+              ></btrix-select-collection-start-page>
+
+              <sl-checkbox name="useThumbnail" class="mt-3" checked>
+                ${msg("Update collection thumbnail")}
+                <sl-tooltip
+                  content=${msg(
+                    "If this collection is public, the preview will be used as the thumbnail for this collection.",
+                  )}
+                >
+                  <sl-icon
+                    name="info-circle"
+                    class="[vertical-align:-.175em]"
+                  ></sl-icon>
+                </sl-tooltip>
+              </sl-checkbox>
+            </section>
+          `,
+        )}
+      </form>
+    `;
+  }
+
+  private renderReplay() {
+    const replaySource = `/api/orgs/${this.orgId}/collections/${this.collectionId}/replay.json`;
+    // TODO Get query from replay-web-page embed
+    const query = queryString.stringify({
+      source: replaySource,
+      customColl: this.collectionId,
+      embed: "default",
+      noCache: 1,
+      noSandbox: 1,
+    });
+
+    return html`<div class="aspect-video w-[200%]">
+      <div class="pointer-events-none aspect-video origin-top-left scale-50">
+        <iframe
+          class="inline-block size-full"
+          src=${`/replay/?${query}#view=pages`}
+        ></iframe>
+      </div>
+    </div>`;
+  }
+
+  private async onSubmit(e: SubmitEvent) {
+    e.preventDefault();
+
+    const form = e.currentTarget as HTMLFormElement;
+    const { homeView, useThumbnail } = serialize(form);
+
+    this.isSubmitting = true;
+
+    try {
+      await this.updateUrl({
+        pageId:
+          (homeView === HomeView.URL && this.selectedSnapshot?.pageId) || null,
+      });
+
+      const shouldUpload =
+        homeView === HomeView.URL &&
+        useThumbnail === "on" &&
+        this.selectedSnapshot &&
+        this.homePageId !== this.selectedSnapshot.pageId;
+      // TODO get filename from rwp?
+      const fileName = `page-thumbnail_${this.selectedSnapshot?.pageId}.jpeg`;
+      let file: File | undefined;
+
+      if (shouldUpload && this.thumbnailPreview?.src) {
+        const { src } = this.thumbnailPreview;
+
+        // Wait to get the thumbnail image before closing the dialog
+        try {
+          const resp = await this.thumbnailPreview.contentWindow!.fetch(src);
+          const blob = await resp.blob();
+
+          file = new File([blob], fileName, {
+            type: blob.type,
+          });
+        } catch (err) {
+          console.debug(err);
+        }
+      } else {
+        this.notify.toast({
+          message: msg("Home view updated."),
+          variant: "success",
+          icon: "check2-circle",
```

Comment:
As of https://github.com/webrecorder/browsertrix/pull/2236 we've got IDs for toasts, so that they can replace rather than stack with other toasts of the same ID:

```suggestion
          icon: "check2-circle",
          id: "home-view-update-status",
```